src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -923,7 +923,10 @@ private static bool IsSimilarOptionButOfHigherRank(List<ReplicationInstruction>
             #region Case 5: Replication and replication guide with type conversion and array promotion
             {
                 //Add as a first attempt a no-replication, but allowing up-promoting
-                replicationTrials.Add(new List<ReplicationInstruction>());
+                if (!replicationTrials.Any(rt => rt.Count == 0))
+                {
+                    replicationTrials.Insert(0, new List<ReplicationInstruction>());
+                }
```

Comment:
This part is actually unrelated. I just noticed that we were adding an empty replication trial even if it was already there. Also, we were adding it to the end of the list, which contradicted the comment, so I changed that as well.